app/views/unattended/provisioning_templates/registration/linux_registration_default.erb

```diff
@@ -20,9 +20,13 @@ set -e
 <%= snippet 'puppet_setup' %>
 <% end -%>
 
+<% if host_param_true?('host_registration_remote_execution') -%>
 <%= snippet 'remote_execution_ssh_keys' %>
-<%= snippet 'insights' -%>
+<% end -%>
 
```

Comment:
A pattern that I would like to see is to correctly handle whitespace
```suggestion

<% end -%>
```

The benefit is that you don't end up with many empty lines. You will also not have to update all snapshots. For end users this makes (p)reviews easier.